lib/castable.h

```diff
@@ -20,40 +20,23 @@ limitations under the License.
 #include <typeinfo>
 
 #include "lib/exceptions.h"
+#include "lib/rtti.h"
 
 /// Handy type conversion methods that can be inherited by various base classes.
-class ICastable {
+class ICastable : public virtual RTTI::Base {
```

Comment:
Please add a comment here that whoever extends this class will also have to declare type info using macros.